Source/EasyNetQ/ChannelDispatcher/PersistentChannelDispatchOptions.cs

```diff
@@ -60,4 +61,28 @@ private ChannelDispatchOptions(string name, PersistentConnectionType connectionT
     ///     True if publisher confirms are enabled
     /// </summary>
     public bool PublisherConfirms { get; }
+
+    protected bool Equals(PersistentChannelDispatchOptions other)
+    {
+        return string.Equals(Name, other.Name, StringComparison.InvariantCulture) && ConnectionType == other.ConnectionType && PublisherConfirms == other.PublisherConfirms;
+    }
+
+    public override bool Equals(object obj)
+    {
+        if (ReferenceEquals(null, obj)) return false;
+        if (ReferenceEquals(this, obj)) return true;
+        if (obj.GetType() != this.GetType()) return false;
+        return Equals((PersistentChannelDispatchOptions)obj);
+    }
+
+    public override int GetHashCode()
+    {
+        unchecked
+        {
+            var hashCode = StringComparer.InvariantCulture.GetHashCode(Name);
+            hashCode = (hashCode * 397) ^ (int)ConnectionType;
+            hashCode = (hashCode * 397) ^ PublisherConfirms.GetHashCode();
+            return hashCode;
+        }
+    }
```

Comment:
Consider remove these methods and switch class to record.

Reply:
Record is seems to be unavailable under netstandard2

Reply:
https://bartwullems.blogspot.com/2021/01/c-9use-record-types-in-net-standard-20.html

Reply:
I saw this hack, but I don't know its consequences, so let's keep as it is now. It is not a big deal to have two autogenerated methods.

Reply:
Technically it is not a hack. It is just how Roslyn compiler works. We can say that this is a documented method of expanding the functionality on old platforms/runtimes. I have already used this well-known trick (not only this particular attribute) in several repositories and over the past time I have not encountered any problems. You will decide.